web/api/badges/web_buffer_svg.c

```diff
@@ -1137,6 +1157,14 @@ int web_client_api_request_v1_badge(RRDHOST *host, struct web_client *w, char *u
     }
 
     cleanup:
+    if(end) {
```

Comment:
Code style, same as above, you get can get rid of `if (end)` and `i =0;` with a for loop.